src/passes/MergeBlocks.cpp

```diff
@@ -491,20 +494,26 @@ struct MergeBlocks : public WalkerPass<PostWalker<MergeBlocks>> {
     return outer;
   }
 
-  void visitUnary(Unary* curr) { optimize(curr, curr->value); }
-  void visitLocalSet(LocalSet* curr) { optimize(curr, curr->value); }
-  void visitLoad(Load* curr) { optimize(curr, curr->ptr); }
-  void visitReturn(Return* curr) { optimize(curr, curr->value); }
+  // Default optimizations for simple cases. Complex things are overridden
+  // below.
```

Comment:
Are we able to override in `UnifiedExpressionVisitor`?

Reply:
Yes. It defines default visitors `visitX() => visitExpression` for all `X`, but a subclass can override them.